components/application_manager/application_manager.h

```diff
@@ -367,7 +367,7 @@ class ApplicationManager {
      *
      * @param app The app to run.
      */
-    void RunApp(std::unique_ptr<App> app);
+    void RunApp(std::unique_ptr<HbbTVApp> app);
```

Comment:
BLOCKING While it makes sense to have the HbbTVApp as a separate class it's now just confusing to use it like this in the application manager. You could define a base class or interface that HbbTVApp and OpApp inherit from and pass this around the ApplicationManager.

Reply:
There is no need at the moment to define an abstract class for those two. OpApp class adds functionality to HbbTV class. There will be duplicate code in OpApp if they both derive from an abstract App class.

Reply:
The idea is to use polymorphism to pass around something called an App. But it now refers explicitly to an HbbTVApp. While the functionality of an OpApp derives from a HbbTVApp it is _not_ a HbbTVApp. So we need to have a way of making this an abstract reference. OpApp can still inherit from HbbTVApp and have a base class.

Reply:
So you are suggesting to OpApp to inherit from two classes?

Reply:
Sorry, realise I've probably confused you completely by asking you to change App class to HbbTVApp. The HbbTVApp is just an app with Hbb specific behaviour and same for OpApp. You can move the common behaviour to a base class that doesn't reference HbbTV or OpApp specifically. Then pass the base class around the ApplicationManager.

Reply:
OK, I will make the change and if it is worth it, I will push it

Reply:
HbbTVApp already provides full functionality, and instances of both HbbTVApp and OpApp can be used interchangeably without ApplicationManager needing to know their exact type. Adding an abstract class would introduce unnecessary complexity without providing additional value.